### PR TITLE
Develop upstream sync 181026

### PR DESCRIPTION
**Pull request by parallelo**

### CODE REVIEW

**Comment by parallelo** [comment]
Initial set of failures on ROCm:  
`Executed 869 out of 869 tests: 850 tests pass and 19 fail locally.`
```
//tensorflow/python:layers_normalization_test_gpu                        [0m[31m[1mFAILED[0m in 79.5s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/layers_normalization_test_gpu/test.log
//tensorflow/python:nn_fused_batchnorm_test_gpu                          [0m[31m[1mFAILED[0m in 34.9s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/nn_fused_batchnorm_test_gpu/test.log
//tensorflow/python/keras:backend_test                                   [0m[31m[1mFAILED[0m in 19.4s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/keras/backend_test/test.log
//tensorflow/python/keras:convolutional_test                             [0m[31m[1mFAILED[0m in 63.4s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/keras/convolutional_test/test.log
//tensorflow/python/kernel_tests:batch_matmul_op_test_gpu                [0m[31m[1mFAILED[0m in 18.4s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/batch_matmul_op_test_gpu/test.log
//tensorflow/python/kernel_tests:cholesky_op_test                        [0m[31m[1mFAILED[0m in 0.2s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/cholesky_op_test/test.log
//tensorflow/python/kernel_tests:cholesky_op_test_gpu                    [0m[31m[1mFAILED[0m in 0.1s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/cholesky_op_test_gpu/test.log
//tensorflow/python/kernel_tests:conv1d_test_gpu                         [0m[31m[1mFAILED[0m in 10.6s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/conv1d_test_gpu/test.log
//tensorflow/python/kernel_tests:conv2d_backprop_filter_grad_test_gpu    [0m[31m[1mFAILED[0m in 22.5s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/conv2d_backprop_filter_grad_test_gpu/test.log
//tensorflow/python/kernel_tests:conv_ops_3d_test_gpu                    [0m[31m[1mFAILED[0m in 9.7s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/conv_ops_3d_test_gpu/test.log
//tensorflow/python/kernel_tests:conv_ops_test_gpu                       [0m[31m[1mFAILED[0m in 15.2s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/conv_ops_test_gpu/test.log
//tensorflow/python/kernel_tests:dct_ops_test_gpu                        [0m[31m[1mFAILED[0m in 9.9s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/dct_ops_test_gpu/test.log
//tensorflow/python/kernel_tests:depthwise_conv_op_test_gpu              [0m[31m[1mFAILED[0m in 41.7s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/depthwise_conv_op_test_gpu/test.log
//tensorflow/python/kernel_tests:fft_ops_test_gpu                        [0m[31m[1mFAILED[0m in 10.8s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/fft_ops_test_gpu/test.log
//tensorflow/python/kernel_tests:list_ops_test_gpu                       [0m[31m[1mFAILED[0m in 12.3s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/list_ops_test_gpu/test.log
//tensorflow/python/kernel_tests:matmul_op_test_gpu                      [0m[31m[1mFAILED[0m in 50.7s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/matmul_op_test_gpu/test.log
//tensorflow/python/kernel_tests:pool_test_gpu                           [0m[31m[1mFAILED[0m in 0.1s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/pool_test_gpu/test.log
//tensorflow/python/kernel_tests:pooling_ops_3d_test_gpu                 [0m[31m[1mFAILED[0m in 9.4s
  /home/jenkins/workspace/tensorflow-upstream-merge-181026/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/k8-opt/testlogs/tensorflow/python/kernel_tests/pooling_ops_3d_test_gpu/test.log
//tensorflow/python/kernel_tests:pooling_ops_test_gpu                    [0m[31m[1mFAILED[0m in 70.3s
```

**Comment by parallelo** [comment]
Updated status --

ROCm:  
```
Executed 869 out of 869 tests: 850 tests pass and 19 fail locally.
```

CUDA:  
```
Executed 867 out of 867 tests: 863 tests pass and 4 fail locally.
```

**Comment by whchung** [comment]
@parallelo any updates on the failed cases? from CI log it seems to have something to do with batch normalization API.

**Comment by parallelo** [comment]
No updates yet, just recently started digging.

**Comment by parallelo** [comment]
Yeah, seeing lots of this:  
```
2018-10-27 15:05:15.068555: I tensorflow/core/graph/gpu_fusion_pass.cc:416] ROCmFusionPass was not enabled!
MIOpen Error: /data/repo/MIOpen/src/ocl/batchnormocl.cpp:65: 
2018-10-27 15:05:15.539676: E tensorflow/stream_executor/rocm/rocm_dnn.cc:2566] failed to enqueue forward batch normalization on stream: miopenStatusBadParm
ERROR:tensorflow:cuDNN launch failure : input shape ([2,2,2,3])
	 [[node batch_normalization/FusedBatchNormV2 (defined at /usr/lib/python3.5/unittest/case.py:600) ]]
	 [[{{node Sum/_7}}]]
```
I'm following up on this with the MIOpen team to get some further clues.  

I also have some misc fixes that I'm about to push out.  Then, will kick off another round of tests to see where we stand.

**Comment by whchung** [comment]
@parallelo CUDA path now working. 17 fails on ROCm path.

Before fixing these issues, perhaps this line in ROCm GPU fusion pass can be disabled so the log is a bit cleaner:
`tensorflow/core/graph/gpu_fusion_pass.cc:416] ROCmFusionPass was not enabled!`

**Comment by whchung** [comment]
Also it seems there are a few more cases which use 5D tensors. Adding @daniellowell to weigh in on the proper APIs be used.

**Comment by daniellowell** [comment]
@whchung We have support for 3D batch norm, tensorOps, copy, set, and scale. However that is about it at this point. No 5-D fusion support.
